feature-libs/organization/user-registration/components/form/user-registration-form.component.ts

```diff
@@ -0,0 +1,92 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { Country, Region } from '@spartacus/core';
+import { Title } from '@spartacus/user/profile/root';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { switchMap, take } from 'rxjs/operators';
+import { UserRegistrationFormService } from './user-registration-form.service';
+
+@Component({
+  selector: 'cx-user-registration-form',
+  templateUrl: './user-registration-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UserRegistrationFormComponent implements OnInit {
+  titles$: Observable<Title[]>;
+
+  countries$: Observable<Country[]>;
+
+  regions$: Observable<Region[]>;
+
+  selectedCountry$: BehaviorSubject<string> = new BehaviorSubject<string>('');
+
+  isLoading$ = new BehaviorSubject(false);
+
+  registerForm: FormGroup;
+
+  messageContent: string;
+
+  constructor(
+    protected userRegistrationFormService: UserRegistrationFormService
+  ) {}
+
+  ngOnInit(): void {
+    this.registerForm = this.userRegistrationFormService.initializeForm();
+
+    this.titles$ = this.userRegistrationFormService.getTitles();
+
+    this.countries$ = this.userRegistrationFormService.getCountries();
+
+    this.regions$ = this.selectedCountry$.pipe(
+      switchMap((country) =>
+        this.userRegistrationFormService.getRegions(country)
+      )
+    );
+  }
+
+  countrySelected(country: Country): void {
+    this.registerForm
+      .get('country')
+      ?.get('isocode')
+      ?.setValue(country?.isocode);
```

Comment:
Is this part necessary? If there was any reason to do so, please remind me. I know the same was done in `address-form.component.ts`, but if you check `unit-address-form.component.ts` it isn't and works well.